File: sanket-dalvi/Day5/twoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, key in enumerate(nums):
            d[key] = i
        for i in range(len(nums)):
            if target-nums[i] in d and nums[i]*2 != target:
                return sorted([i, d[target-nums[i]]])
        for i in range(len(nums)):
            if nums.count(nums[i]) > 1:
                a = nums.index(nums[i])
                b = nums[a+1:].index(nums[i])+a+1
                return sorted([a,b])


# Test
print(Solution().twoSum(nums = [2,7,11,15], target = 9))

File: parasmittal/Day 5/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
     unordered_map<int,int> um;
        vector<int> o;
        int n=nums.size();
        for(int i=0;i<n;i++)
            um[nums[i]]=i;
        for(int i=0;i<n;i++){
            int x=target-nums[i];
            if(um.find(x)!=um.end() && i!=um[x]){
                o.push_back(i);
                o.push_back(um[x]);
                return o;
        }
    }
        return o;
        }
};